From: L. David Baron <dbaron@dbaron.org>

Bug 864686:  Make reftest harness print test URL when there is one.

I think the case in this bug is the RecvException case.

diff --git a/layout/tools/reftest/reftest.js b/layout/tools/reftest/reftest.js
--- a/layout/tools/reftest/reftest.js
+++ b/layout/tools/reftest/reftest.js
@@ -1499,17 +1499,17 @@ function RecordResult(testRunTime, error
         return;
     }
 
     if (gURLs[0]["prefSettings" + gState].length == 0 &&
         gURICanvases[gCurrentURL]) {
         gCurrentCanvas = gURICanvases[gCurrentURL];
     }
     if (gCurrentCanvas == null) {
-        gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | | program error managing snapshots\n");
+        gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | " + gCurrentURL + " | program error managing snapshots\n");
         ++gTestResults.Exception;
     }
     if (gState == 1) {
         gCanvas1 = gCurrentCanvas;
     } else {
         gCanvas2 = gCurrentCanvas;
     }
     gCurrentCanvas = null;
@@ -1812,17 +1812,17 @@ function RecvAssertionCount(count)
 function RecvContentReady()
 {
     InitAndStartRefTests();
     return { remote: gBrowserIsRemote };
 }
 
 function RecvException(what)
 {
-    gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | | "+ what +"\n");
+    gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | " + gCurrentURL + " | " + what + "\n");
     ++gTestResults.Exception;
 }
 
 function RecvFailedLoad(why)
 {
     LoadFailed(why);
 }
 
@@ -1835,17 +1835,17 @@ function RecvInitCanvasWithSnapshot()
 function RecvLog(type, msg)
 {
     msg = "[CONTENT] "+ msg;
     if (type == "info") {
         LogInfo(msg);
     } else if (type == "warning") {
         LogWarning(msg);
     } else {
-        gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | | unknown log type "+ type +"\n");
+        gDumpLog("REFTEST TEST-UNEXPECTED-FAIL | " + gCurrentURL + " | unknown log type " + type + "\n");
         ++gTestResults.Exception;
     }
 }
 
 function RecvScriptResults(runtimeMs, error, results)
 {
     RecordResult(runtimeMs, error, results);
 }
